internal/rpmmd/repository.go

```diff
@@ -186,6 +186,13 @@ type DNFError struct {
 	Reason string `json:"reason"`
 }
 
+// DNFException inherits from DNFError. Represents an error due to a crash
+// on the python side. Its purpose is to give a way to the user of Depsolve to
+// differentiate the types of errors.
+type DNFException struct {
```

Comment:
Hmm, all DNFErrors are `dnf.exceptions`. We should filter what to do with the based on `DNFError.Kind` rather than the golang type.

Reply:
Yeah, I know, using type overloading to sort the exceptions was not the most straightforward approach, but this was mainly to avoid string parsing.
And in fact, in the end, this type does not proves itself this useful.